electrumx/server/block_processor.py

```diff
@@ -534,6 +534,9 @@ def advance_txs(self, txs):
             # Spend the inputs
             if not tx.is_coinbase:
                 for txin in tx.inputs:
+                    # Verify that this is not a stealth input
+                    if self.coin.txin_is_spendable(txin) is False:
+                        continue
```

Comment:
Sorry adding coin-specific code to this key loop is not acceptable.  But I don't see why you need to customise this.

Reply:
I'd like you to explain why an input is treated this way and what it means.  The input that you're ignoring and not allowing to be spent remains in the UTXO set?  Or was it never there in the first place?   In which case what do the prev tx_hash and prev_index mean?

Reply:
So the input we are ignoring is actually a stealth input. Essentially, when someone sends a certain amount of coin to a stealth address, those coins get "burned". They get removed from the public ledger and added to the private one.

The reverse also happens when sending coins from a private address to a normal address. They essentially get minted in the public ledger. For these types of transactions, there is no ```prev_index``` to point to. Electrumx being a public index, I think it's safe to ignore these inputs, and only count any public outputs. This also happens in the reverse case. We ignore "invalid" (OP_RETURN) outputs, when a normal input sends coins to a stealth address.

Here is a high level (stratospheric) explanation of what happens:

https://github.com/tokenpay/tokenpay/blob/master/doc/Ring%20Signatures.md